### PR TITLE
Clean up poddisruptionbudget of non-existing instance manager pods

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#1944

### CODE REVIEW

File: controller/instance_manager_controller.go [hunk 1]
```diff
@@ -265,6 +265,11 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 	if err != nil {
 		if datastore.ErrorIsNotFound(err) {
 			logrus.Infof("Longhorn instance manager %v has been deleted", key)
+			if err := imc.ds.DeletePDB(getPDBNameFromIMName(name)); err != nil {
```
Comment (yasker) [inline controller/instance_manager_controller.go:268]: It's possible for us to miss this event. I don't think we should depend on the deletion notification. It's better to compare what we have set right now vs what's need to be set right now.

Comment (PhanLe1010) [reply controller/instance_manager_controller.go:268]: I see. Thank you!

Comment (PhanLe1010) [reply controller/instance_manager_controller.go]: Fixed. Thanks

**Comment by PhanLe1010** [comment]
E2E tests doesn't pass. I am looking into it. `test_expansion_canceling `

**Comment by PhanLe1010** [comment]
Update: E2E core tests passed except for `test_ha_salvage` which is a known problem.

**Review by yasker — APPROVED** [review]

**Pull request merged**